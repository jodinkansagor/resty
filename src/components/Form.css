form {
  font-family: 'Source Sans Pro', sans-serif;
  background-color: #eeeeee;
  color: #0b0c10;
  margin-top: 5px;
  padding-bottom: 5px;
  

  .jsonAndAuth {
    display: flex;
    flex-direction: row;

    input[type=text] {
      margin: 2px;
    }
  }

  .url {
    margin: 5px;
    margin-bottom: 20px;
    width: 98%;
  }

  .radio {
    display: none;
  }
  
  input[type=radio]+span {
    padding: 7px;
  }

  input[type=radio]:checked+span {
    background-color: #45a29e;
    color: #c5c6c7
  }

  input[type=radio]:hover+span {
    background-color: #c5c6c7;
  }
  
  label {
    cursor: pointer;
    margin: 10px;
  }

  button {
    cursor: pointer;
  }
  
  textarea {
    margin-top: 15px;
    margin-right: 15px;
    margin-left: 5px;
  }
}