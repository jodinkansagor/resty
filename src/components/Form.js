import React from 'react';
import PropTypes from 'prop-types';
import styles from '../components/Form.css';



const Form = ({ onSubmit, url, textInput, onChange, userName, password, bearerToken }) => (

  <form onSubmit={onSubmit}>
    <input type="text" className={styles.url} placeholder="URL" value={url} name="url" onChange={onChange} />
    <section className={styles.radioButtons}>
      <label><input className={styles.radio} type="radio" name="method" value="GET" onChange={onChange}></input><span>GET</span></label>
      <label><input className={styles.radio} type="radio" name="method" value="POST" onChange={onChange}></input><span>POST</span></label>
      <label><input className={styles.radio} type="radio" name="method" value="PATCH" onChange={onChange}></input><span>PATCH</span></label>
      <label><input className={styles.radio} type="radio" name="method" value="PUT" onChange={onChange}></input><span>PUT</span></label>
      <label><input className={styles.radio} type="radio" name="method" value="DELETE" onChange={onChange}></input><span>DELETE</span></label>
      <button>GO</button>
    </section>
    <section className={styles.jsonAndAuth}>
      <textarea name="textInput" type="text" value={textInput} placeholder="Raw JSON" onChange={onChange}></textarea>
      <section className={styles.justAuth}>
        <h2>Basic Authorization</h2>
        <input type="text" placeholder="Username" value={userName} name="userName" onChange={onChange} />
        <input type="text" placeholder="Password" value={password} name="password" onChange={onChange} />
        <h2>Bearer Token</h2>
        <input type="text" placeholder="Bearer Token" value={bearerToken} name="bearerToken" onChange={onChange} />
      </section>
    </section>
  </form>
);

Form.propTypes = {
  url: PropTypes.string.isRequired,
  userName: PropTypes.string.isRequired,
  password: PropTypes.string.isRequired,
  bearerToken: PropTypes.string.isRequired,
  textInput: PropTypes.string.isRequired,
  onSubmit: PropTypes.func.isRequired,
  onChange: PropTypes.func.isRequired
};

export default Form;
