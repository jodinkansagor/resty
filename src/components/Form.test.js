import React from 'react';
import { shallow } from 'enzyme';
import Form from './Form';

describe('Form component', () => {
  it('renders Form', () => {
    const wrapper = shallow(<Form method="method" url="url" userName="username" password="password" bearerToken="bearerToken" textInput="textInput" onChange={()=>{}} onSubmit={()=>{}} />);
    expect(wrapper).toMatchSnapshot();
  });
});
