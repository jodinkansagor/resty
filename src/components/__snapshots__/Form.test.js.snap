// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form component renders Form 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    className="url"
    name="url"
    onChange={[Function]}
    placeholder="URL"
    type="text"
    value="url"
  />
  <section
    className="radioButtons"
  >
    <label>
      <input
        checked={false}
        className="radio"
        name="method"
        onChange={[Function]}
        type="radio"
        value="GET"
      />
      <span>
        GET
      </span>
    </label>
    <label>
      <input
        checked={false}
        className="radio"
        name="method"
        onChange={[Function]}
        type="radio"
        value="POST"
      />
      <span>
        POST
      </span>
    </label>
    <label>
      <input
        checked={false}
        className="radio"
        name="method"
        onChange={[Function]}
        type="radio"
        value="PATCH"
      />
      <span>
        PATCH
      </span>
    </label>
    <label>
      <input
        checked={false}
        className="radio"
        name="method"
        onChange={[Function]}
        type="radio"
        value="PUT"
      />
      <span>
        PUT
      </span>
    </label>
    <label>
      <input
        checked={false}
        className="radio"
        name="method"
        onChange={[Function]}
        type="radio"
        value="DELETE"
      />
      <span>
        DELETE
      </span>
    </label>
    <button>
      GO
    </button>
  </section>
  <section
    className="jsonAndAuth"
  >
    <textarea
      name="textInput"
      onChange={[Function]}
      placeholder="Raw JSON"
      type="text"
      value="textInput"
    />
    <section
      className="justAuth"
    >
      <h2>
        Basic Authorization
      </h2>
      <input
        name="userName"
        onChange={[Function]}
        placeholder="Username"
        type="text"
        value="username"
      />
      <input
        name="password"
        onChange={[Function]}
        placeholder="Password"
        type="text"
        value="password"
      />
      <h2>
        Bearer Token
      </h2>
      <input
        name="bearerToken"
        onChange={[Function]}
        placeholder="Bearer Token"
        type="text"
        value="bearerToken"
      />
    </section>
  </section>
</form>
`;
